examples/basic_mysql_connection/index.js

```diff
@@ -4,17 +4,28 @@ var mysql = require('mysql');
 var config = JSON.parse(fs.readFileSync("config-secret.json"))
 
 var connection = mysql.createConnection({
-	host: config.host,
+	 host: config.host,
 	user: config.user,
 	password: config.password,
 	port: config.port,
-    database: config.database
+  database: config.database
 });
 
 connection.connect();
+const sqlStatments=['SELECT COUNT(*) as "Total No of Todos" FROM todos',
+					'SELECT * FROM todos WHERE Done=1',
+					'SELECT * FROM todos WHERE Done=0',
+					'SELECT * FROM todos GROUP BY Due DESC',
+					'SELECT * FROM todos WHERE Id=(SELECT MAX(Id) FROM todos) ',
```

Comment:
You did something very similar to Amir so I'm going to copy my comments to him here, and then add some more for you:

<hr>

This is an interesting use of a subquery, and not necessarily how I would choose to write this. The question isn't super clear, so I apologise for that.

However, subqueries can be dramatically slower than other forms of the same query. Can you suggest another way this could be written which doesn't require a subquery?

For an example of why a subquery would be slower, imagine that this table has a billion rows. The subquery gets executed first, which will involve reading all one billion rows to find the max. Then, the main query will execute, which involves scanning all one billion rows again to find the match.

<hr>

You're very close to what I would consider the 'correct' solution (but there is more than one way). Can you try and come up with another query which returns the same result but without using a subquery?